### PR TITLE
Release notes: add info about segwit and null dummy soft forks

### PR DESCRIPTION
**Pull request by harding**
- Segwit text primarily from https://bitcoincore.org/en/2016/06/24/segwit-next-steps/
- null dummy text partially from BIP147

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -38,8 +38,82 @@ report issues about Windows XP to the issue tracker.
...
+  new-style blocks can hold more data than older-style blocks, allowing
+  a modest increase to the amount of transaction data that can fit in a
+  block.
+
+Segwit also simplifies the ability to add new features to Bitcoin and
+improves the efficiency of full nodes, which will help provide long-term
+benefits to Bitcoin users.
+
+Activation for the segwit soft fork is being managed using BIP9 versionbits.
+The earliest activation date is set to 15 November 2016 (UTC) and
```
Comment (sipa) [inline doc/release-notes.md]: That's not technically correct, as it requires a signalling period and a fallow period (both 2016 blocks) after the start time (unless you assume 4032 blocks can be mined within 24 hours).

Comment (harding) [inline doc/release-notes.md]: Oh, indeed.  What was I thinking.  I'll edit.

Comment (btcdrak) [inline doc/release-notes.md]: Nov 15th is the earliest date that signalling can start. Earliest activation is likely to be ~28 days/4032 blocks from the next retarget after 15th Nov (estimated to be 20th Nov). I would hazard to be so specific.

File: doc/release-notes.md [hunk 2]
```diff
@@ -66,9 +66,12 @@ Segwit also simplifies the ability to add new features to Bitcoin and
 improves the efficiency of full nodes, which will help provide long-term
 benefits to Bitcoin users.
 
-Activation for the segwit soft fork is being managed using BIP9 versionbits.
-The earliest activation date is set to 15 November 2016 (UTC) and
-its versionbit is bit 1.
+Activation for the segwit soft fork is being managed using BIP9
+versionbits.  Segwit's version bit is bit 1, and nodes will begin
+tracking which blocks signal support for segwit on 15 November 2016
```
Comment (btcdrak) [inline doc/release-notes.md]: Still not technically correct. Tracking will start from the first retarget after the 15th.

**Comment by harding** [comment]
(Please feel free to squash at merge time, or let me know and I'll squash everything on this branch.)

**Comment by laanwj** [comment]
Thanks for writing this!

ACK

**Comment by btcdrak** [comment]
ACK

**Pull request merged**